classes/Events/SaveNotificationOnMessageSaveEvent.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+namespace Avado\MoodleAbstractionLibrary\Events;
+
+use Avado\MoodleAbstractionLibrary\Entities\Notification;
+use Avado\MoodleAbstractionLibrary\Entities\NotificationComponent;
+use Avado\MoodleAbstractionLibrary\Entities\NotificationEmailQueue;
+use Avado\MoodleAbstractionLibrary\Entities\NotificationEventType;
+use Avado\MoodleAbstractionLibrary\Entities\NotificationSetting;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\QueryException;
+
+/**
+ * Class SaveNotificationOnMessageSaveEvent
+ * @package Avado\MoodleAbstractionLibrary\Events\AbstractEvent
+ */
+class SaveNotificationOnMessageSaveEvent extends AbstractEvent
+{
+    /**
+     * @var bool
+     */
+    public const QUEUED = false;
+
+    /**
+     * @param array $arguments
+     * @return bool
+     */
+    public function check(array $arguments): bool
+    {
+        return true;
```

Comment:
Just a suggestion,
Within check function can we check validation like

$notificationSettings = $this->notificationSettings($message->useridto, $eventTypeId);

 if($notificationSettings->isNotEmpty()) {
 return true;
 }

Reply:
Thats a good point and something I considered but left based on the below. 

As this is a hopefully short term patch until we move other functionality away from using the Message table It will always run instantly, and be redundant by the time we implement queuing. Therefor having the check in the execute function as opposed to 'check()' itself saves on duplicate DB queries. 

Although if you feel that's not good reasoning I can change it?

Reply:
Ok Jon